### PR TITLE
Fix tests and lint; Update .gitignore

### PR DESCRIPTION
**Pull request by StK88**

### CODE REVIEW

**Comment by AnrDaemon** [comment]
Tests do not run for me. Fail with

```
> branch-in-window-title@0.1.4-1 test
> node ./out/test/runTest.js

Downloading VS Code 1.88.1 from https://update.code.visualstudio.com/1.88.1/win32-archive/stable
Failed to get VS Code archive location
Failed to run tests
```

Trying the same URL in the browser yields `404 Not found` HTTP error.

**Comment by StK88** [comment]
The above issue is from the original repo and relates only to Windows.
Can you pull a fix, run `npm ci`, and rerun tests?

File: package.json [hunk 1]
```diff
@@ -68,12 +68,12 @@
...
 		"@typescript-eslint/eslint-plugin": "^5.18.0",
 		"@typescript-eslint/parser": "^5.18.0",
+		"@vscode/test-electron": "^2.3.9",
 		"eslint": "^8.12.0",
 		"glob": "^7.2.0",
 		"mocha": "^9.2.2",
 		"sinon": "^17.0.1",
 		"text-encoding": "^0.7.0",
-		"typescript": "^4.6.3",
-		"vscode-test": "^1.6.1"
```
Comment (StK88) [inline package.json:75]: [vscode-test](https://www.npmjs.com/package/vscode-test) is deprecated , replaced with [@vscode/test-electron](https://www.npmjs.com/package/@vscode/test-electron). Fixes issue on Windows.

File: src/test/runTest.ts [hunk 2]
```diff
@@ -13,7 +14,11 @@ async function main() {
 		const extensionTestsPath = path.resolve(__dirname, './suite/index');
 
 		// Download VS Code, unzip it and run the integration test
-		await runTests({ extensionDevelopmentPath, extensionTestsPath });
+		await runTests({
+			extensionDevelopmentPath,
+			extensionTestsPath,
+			launchArgs: ['--user-data-dir', `${os.tmpdir()}`],
```
Comment (StK88) [inline src/test/runTest.ts:20]: Fixes issue on macOS when tests run twice in a row and throw error: `Could not delete obsolete instance`.

**Comment by AnrDaemon** [comment]
Test suite fails intermittently on consecutive runs.
But the failing test is always `branchDidChange should be called with branchNameFromGit`. Which is weird when it happens in the title tests.

```
 *  Executing task: npm run test 


> branch-in-window-title@0.1.4-1 pretest
> npm run compile && npm run lint       


> branch-in-window-title@0.1.4-1 compile
> tsc -p ./


> branch-in-window-title@0.1.4-1 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4-1 test
> node ./out/test/runTest.js

Downloading VS Code 1.88.1 from https://update.code.visualstudio.com/1.88.1/win32-x64-archive/stable
Downloading VS Code [==============================] 100%
Downloaded VS Code into C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1

Downloaded VS Code into C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1

[main 2024-04-30T18:05:29.913Z] Error: Error mutex already exists
    at dt.W (C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1\resources\app\out\vs\code\electron-main\main.js:110:68581)
[main 2024-04-30T18:05:29.916Z] update#setState disabled
[main 2024-04-30T18:05:29.916Z] update#ctor - updates are disabled by the environment
Started initializing default profile extensions in extensions installation folder. file:///c%3A/Users/anrdaemon/Documents/.github/branch-in-window-title/.vscode-test/extensions
Started local extension host with pid 13192.
Completed initializing default profile extensions in extensions installation folder. file:///c%3A/Users/anrdaemon/Documents/.github/branch-in-window-title/.vscode-test/extensions
Loading development extension at c:\Users\anrdaemon\Documents\.github\branch-in-window-title

  Branch Detector Integration Test Suite
    From project root path
      ✔ branchDidChange should be called with branchNameFromGit (65ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      ✔ branchDidChange should be called with branchNameFromGit (67ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    ✔ updateTitle should append branch information to title
    ✔ updateTitle should prepend branch information to title
    ✔ updateTitle should remove branch information from title if branch is undefined
    ✔ updateTitle should not call set title operation if the title is unchanged
    ✔ updateTitle should not alter the title if template does not match
  13 passing (237ms)
[main 2024-04-30T18:05:34.706Z] Extension host with pid 13192 exited with code: 0, signal: unknown.
Exit code:   0
Done

 *  Terminal will be reused by tasks, press any key to close it. 

 *  Executing task: npm run test 


> branch-in-window-title@0.1.4-1 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4-1 compile
> tsc -p ./


> branch-in-window-title@0.1.4-1 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4-1 test
> node ./out/test/runTest.js

Found existing install in C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1. Skipping download

[main 2024-04-30T18:07:00.365Z] Error: Error mutex already exists
    at dt.W (C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1\resources\app\out\vs\code\electron-main\main.js:110:68581)
[main 2024-04-30T18:07:00.368Z] update#setState disabled
[main 2024-04-30T18:07:00.368Z] update#ctor - updates are disabled by the environment
Started local extension host with pid 14120.
Loading development extension at c:\Users\anrdaemon\Documents\.github\branch-in-window-title

  Branch Detector Integration Test Suite
    From project root path
      ✔ branchDidChange should be called with branchNameFromGit (77ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      1) branchDidChange should be called with branchNameFromGit
      ✔ branchDidChange should be called with branchNameFromGit (149ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    ✔ updateTitle should append branch information to title
    2) branchDidChange should be called with branchNameFromGit
    ✔ updateTitle should prepend branch information to title
    ✔ updateTitle should remove branch information from title if branch is undefined
    ✔ updateTitle should not call set title operation if the title is unchanged
rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: A branch is detected
stack trace: AssertionError [ERR_ASSERTION]: A branch is detected
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:49:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: No branch detected
stack trace: AssertionError [ERR_ASSERTION]: No branch detected
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:56:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: No branch detected
stack trace: AssertionError [ERR_ASSERTION]: No branch detected
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:64:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    ✔ updateTitle should not alter the title if template does not match
rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: No branch detected
stack trace: AssertionError [ERR_ASSERTION]: No branch detected
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:96:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: A branch is detected
stack trace: AssertionError [ERR_ASSERTION]: A branch is detected
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:81:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: No branch detected
stack trace: AssertionError [ERR_ASSERTION]: No branch detected
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:88:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
  13 passing (2s)
  2 failing
  1) Branch Detector Integration Test Suite
       From project root path
         branchDidChange should be called with branchNameFromGit:
     Error: done() called multiple times in test <Branch Detector Integration Test Suite From project root path branchDidChange should be called with branchNameFromGit> of file c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:42:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99

  2) Branch Detector Integration Test Suite
       From project sub folder path
         branchDidChange should be called with branchNameFromGit:
     Error: done() called multiple times in test <Branch Detector Integration Test Suite From project sub folder path branchDidChange should be called with branchNameFromGit> of file c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:74:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99

Error: 2 tests failed.
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\index.js:25:27
        at done (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\node_modules\mocha\lib\mocha.js:1058:7)       
[main 2024-04-30T18:07:05.849Z] Extension host with pid 14120 exited with code: 0, signal: unknown.
Exit code:   1
Failed to run tests

 *  The terminal process "C:\Programs\Cygwin\bin\bash.exe '-i', '-c', 'npm run test'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

 *  Executing task: npm run test 


> branch-in-window-title@0.1.4-1 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4-1 compile
> tsc -p ./


> branch-in-window-title@0.1.4-1 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4-1 test
> node ./out/test/runTest.js

Found existing install in C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1. Skipping download

[main 2024-04-30T18:07:25.914Z] Error: Error mutex already exists
    at dt.W (C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1\resources\app\out\vs\code\electron-main\main.js:110:68581)
[main 2024-04-30T18:07:25.917Z] update#setState disabled
[main 2024-04-30T18:07:25.917Z] update#ctor - updates are disabled by the environment
Started local extension host with pid 15356.
Loading development extension at c:\Users\anrdaemon\Documents\.github\branch-in-window-title

  Branch Detector Integration Test Suite
    From project root path
      ✔ branchDidChange should be called with branchNameFromGit (87ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      ✔ branchDidChange should be called with branchNameFromGit (81ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    ✔ updateTitle should append branch information to title
    ✔ updateTitle should prepend branch information to title
    1) branchDidChange should be called with branchNameFromGit
    ✔ updateTitle should remove branch information from title if branch is undefined
    ✔ updateTitle should not call set title operation if the title is unchanged
    ✔ updateTitle should not alter the title if template does not match
Error: 1 tests failed.
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\index.js:25:27
        at done (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\node_modules\mocha\lib\mocha.js:1058:7)       
  13 passing (360ms)
  1 failing
  1) Branch Detector Integration Test Suite
       From project root path
         branchDidChange should be called with branchNameFromGit:
     Error: done() called multiple times in test <Branch Detector Integration Test Suite From project root path branchDidChange should be called with branchNameFromGit> of file c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\test\suite\branchDetector.test.js:42:17
        at BranchDetector.callIfChanged (c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:36:13)
        at c:\Users\anrdaemon\Documents\.github\branch-in-window-title\out\branchDetector.js:18:99
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

[main 2024-04-30T18:07:29.166Z] Extension host with pid 15356 exited with code: 0, signal: unknown.
Exit code:   1
Failed to run tests

 *  The terminal process "C:\Programs\Cygwin\bin\bash.exe '-i', '-c', 'npm run test'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

 *  Executing task: npm run test 


> branch-in-window-title@0.1.4-1 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4-1 compile
> tsc -p ./


> branch-in-window-title@0.1.4-1 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4-1 test
> node ./out/test/runTest.js

Found existing install in C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1. Skipping download

[main 2024-04-30T18:07:57.216Z] Error: Error mutex already exists
    at dt.W (C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1\resources\app\out\vs\code\electron-main\main.js:110:68581)
[main 2024-04-30T18:07:57.219Z] update#setState disabled
[main 2024-04-30T18:07:57.219Z] update#ctor - updates are disabled by the environment
Started local extension host with pid 14400.
Loading development extension at c:\Users\anrdaemon\Documents\.github\branch-in-window-title

  Branch Detector Integration Test Suite
    From project root path
      ✔ branchDidChange should be called with branchNameFromGit (72ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      ✔ branchDidChange should be called with branchNameFromGit (72ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    ✔ updateTitle should append branch information to title
    ✔ updateTitle should prepend branch information to title
    ✔ updateTitle should remove branch information from title if branch is undefined
    ✔ updateTitle should not call set title operation if the title is unchanged
    ✔ updateTitle should not alter the title if template does not match
  13 passing (283ms)
[main 2024-04-30T18:08:00.288Z] Extension host with pid 14400 exited with code: 0, signal: unknown.
Exit code:   0
Done

 *  Terminal will be reused by tasks, press any key to close it. 

 *  Executing task: npm run test 


> branch-in-window-title@0.1.4-1 pretest
> npm run compile && npm run lint       


> branch-in-window-title@0.1.4-1 compile
> tsc -p ./


> branch-in-window-title@0.1.4-1 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4-1 test
> node ./out/test/runTest.js

Found existing install in C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1. Skipping download

[main 2024-04-30T18:09:25.232Z] Error: Error mutex already exists
    at dt.W (C:\Users\anrdaemon\Documents\.github\branch-in-window-title\.vscode-test\vscode-win32-x64-archive-1.88.1\resources\app\out\vs\code\electron-main\main.js:110:68581)
[main 2024-04-30T18:09:25.235Z] update#setState disabled
[main 2024-04-30T18:09:25.235Z] update#ctor - updates are disabled by the environment
Loading development extension at c:\Users\anrdaemon\Documents\.github\branch-in-window-title
Started local extension host with pid 14820.

  Branch Detector Integration Test Suite
    From project root path
      ✔ branchDidChange should be called with branchNameFromGit (71ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      ✔ branchDidChange should be called with branchNameFromGit (71ms)
      ✔ branchDidChange should be called with mocked git branch name
      ✔ branchDidChange should be called with undefined when mocked git branch is undefined
      ✔ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    ✔ updateTitle should append branch information to title
    ✔ updateTitle should prepend branch information to title
    ✔ updateTitle should remove branch information from title if branch is undefined
    ✔ updateTitle should not call set title operation if the title is unchanged
    ✔ updateTitle should not alter the title if template does not match
  13 passing (264ms)
[main 2024-04-30T18:09:28.555Z] Extension host with pid 14820 exited with code: 0, signal: unknown.
Exit code:   0
Done

 *  Terminal will be reused by tasks, press any key to close it. 
```

**Comment by StK88** [comment]
I couldn't reproduce it on my Windows machine, here are 4 consecutive runs:
```
st@stk /cygdrive/d/work/vscode-branch-in-window-title-my-fork
$ npm run test

> branch-in-window-title@0.1.4 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4 compile
> tsc -p ./


> branch-in-window-title@0.1.4 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4 test
> node ./out/test/runTest.js

Found existing install in D:\work\vscode-branch-in-window-title-my-fork\.vscode-
test\vscode-win32-x64-archive-1.88.1. Skipping download

[main 2024-05-01T08:04:23.547Z] update#setState disabled
[main 2024-05-01T08:04:23.548Z] update#ctor - updates are disabled by the enviro
nment
Started local extension host with pid 9116.
Loading development extension at d:\work\vscode-branch-in-window-title-my-fork

  Branch Detector Integration Test Suite
    From project root path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch i
s undefined
      √ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch i
s undefined
      √ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    √ updateTitle should append branch information to title
    √ updateTitle should prepend branch information to title
    √ updateTitle should remove branch information from title if branch is undef
ined
    √ updateTitle should not call set title operation if the title is unchanged
    √ updateTitle should not alter the title if template does not match
  13 passing (105ms)
[main 2024-05-01T08:04:27.284Z] Extension host with pid 9116 exited with code: 0
, signal: unknown.
Exit code:   0
Done


st@stk /cygdrive/d/work/vscode-branch-in-window-title-my-fork
$ npm run test

> branch-in-window-title@0.1.4 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4 compile
> tsc -p ./


> branch-in-window-title@0.1.4 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4 test
> node ./out/test/runTest.js

Found existing install in D:\work\vscode-branch-in-window-title-my-fork\.vscode-
test\vscode-win32-x64-archive-1.88.1. Skipping download

[main 2024-05-01T08:04:36.895Z] update#setState disabled
[main 2024-05-01T08:04:36.896Z] update#ctor - updates are disabled by the enviro
nment
Started local extension host with pid 3944.
Loading development extension at d:\work\vscode-branch-in-window-title-my-fork

  Branch Detector Integration Test Suite
    From project root path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch i
s undefined
      √ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch i
s undefined
      √ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    √ updateTitle should append branch information to title
    √ updateTitle should prepend branch information to title
    √ updateTitle should remove branch information from title if branch is undef
ined
    √ updateTitle should not call set title operation if the title is unchanged
    √ updateTitle should not alter the title if template does not match
  13 passing (114ms)
[main 2024-05-01T08:04:38.155Z] Extension host with pid 3944 exited with code: 0
, signal: unknown.
Exit code:   0
Done


st@stk /cygdrive/d/work/vscode-branch-in-window-title-my-fork
$ npm run test

> branch-in-window-title@0.1.4 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4 compile
> tsc -p ./


> branch-in-window-title@0.1.4 lint
> eslint src --ext ts



> branch-in-window-title@0.1.4 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4 compile
> tsc -p ./


> branch-in-window-title@0.1.4 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4 test
> node ./out/test/runTest.js

Found existing install in D:\work\vscode-branch-in-window-title-my-fork\.vscode-test\vscode-w
in32-x64-archive-1.88.1. Skipping download

[main 2024-05-01T08:04:58.596Z] update#setState disabled
[main 2024-05-01T08:04:58.597Z] update#ctor - updates are disabled by the environment
Started local extension host with pid 3128.
Loading development extension at d:\work\vscode-branch-in-window-title-my-fork

  Branch Detector Integration Test Suite
    From project root path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch is undefined
      √ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch is undefined
      √ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    √ updateTitle should append branch information to title
    √ updateTitle should prepend branch information to title
    √ updateTitle should remove branch information from title if branch is undefined
    √ updateTitle should not call set title operation if the title is unchanged
    √ updateTitle should not alter the title if template does not match
  13 passing (113ms)
[main 2024-05-01T08:04:59.863Z] Extension host with pid 3128 exited with code: 0, signal: unk
nown.
Exit code:   0
Done


st@stk /cygdrive/d/work/vscode-branch-in-window-title-my-fork
$ npm run test

> branch-in-window-title@0.1.4 pretest
> npm run compile && npm run lint


> branch-in-window-title@0.1.4 compile
> tsc -p ./


> branch-in-window-title@0.1.4 lint
> eslint src --ext ts


> branch-in-window-title@0.1.4 test
> node ./out/test/runTest.js

Found existing install in D:\work\vscode-branch-in-window-title-my-fork\.vscode-test\vscode-w
in32-x64-archive-1.88.1. Skipping download

[main 2024-05-01T08:05:29.454Z] update#setState disabled
[main 2024-05-01T08:05:29.455Z] update#ctor - updates are disabled by the environment
Started local extension host with pid 5080.
Loading development extension at d:\work\vscode-branch-in-window-title-my-fork

  Branch Detector Integration Test Suite
    From project root path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch is undefined
      √ branchDidChange should be called with undefined when exception is thrown
    From project sub folder path
      √ branchDidChange should be called with branchNameFromGit
      √ branchDidChange should be called with mocked git branch name
      √ branchDidChange should be called with undefined when mocked git branch is undefined
      √ branchDidChange should be called with undefined when error is thrown
  Title Updater Test Suite
    √ updateTitle should append branch information to title
    √ updateTitle should prepend branch information to title
    √ updateTitle should remove branch information from title if branch is undefined
    √ updateTitle should not call set title operation if the title is unchanged
    √ updateTitle should not alter the title if template does not match
  13 passing (116ms)
[main 2024-05-01T08:05:30.725Z] Extension host with pid 5080 exited with code: 0, signal: unk
nown.
Exit code:   0
Done
```

I'm not sure if that will help, you can try to remove `out/` tests folder manually and run `npm run test` again so it will regenerate fresh `out/`, maybe there's some cache from old tests.

**Comment by AnrDaemon** [comment]
I did a full cleanup. Will try updating node and see if it'd help.

**Comment by AnrDaemon** [comment]
I think I've found a workaround.

**Comment by AnrDaemon** [comment]
Part of the answer was in the https://stackoverflow.com/questions/47252219/event-based-unit-tests-fail-with-done-called-multiple-times

As far as I understand it, the very first `branchDetector()` call creates a disposable, which is still running through the rest of the testing suite, calling update callback over and over again. And when that callback races with an actual test callback, error happens.

Explicitly disposing it in the success callback cleans the area up.

**Comment by StK88** [comment]
Interesting, what `node --version` do you use? Mine is `18.16.0`, maybe it occurs only on lower node versions.

**Comment by AnrDaemon** [comment]
```
$ node --version
v18.15.0
```

**Comment by AnrDaemon** [comment]
Though I distinctly remember running 18.20 installer to completion…

Rechecked with 18.20 now, same issue remains if I commend out detector disposing.

**Comment by StK88** [comment]
I see, thanks for testing it out, then less likely related to node version.

**Pull request merged**